runtime/src/bank.rs

```diff
@@ -5373,32 +5378,47 @@ impl Bank {
                         self.rc
                             .accounts
                             .load_to_collect_rent_eagerly(&self.ancestors, subrange)
-                    };
-                    load.stop();
-                    metrics.load_us.fetch_add(load.as_us(), Relaxed);
-
-                    self.collect_rent_from_accounts(accounts, &metrics, just_rewrites);
-                    metrics
+                    });
+                    let info = self.collect_rent_from_accounts(accounts, just_rewrites);
+                    CollectRentInPartitionInfo::new(
+                        info,
+                        Duration::from_nanos(measure_load_accounts.as_ns()),
+                    )
                 })
-                .collect::<Vec<_>>()
-            {
-                metrics
-                    .load_us
-                    .fetch_add(thread_metrics.load_us.load(Relaxed), Relaxed);
-                metrics
-                    .store_us
-                    .fetch_add(thread_metrics.store_us.load(Relaxed), Relaxed);
-                metrics
-                    .hash_us
-                    .fetch_add(thread_metrics.hash_us.load(Relaxed), Relaxed);
-                metrics
-                    .collect_us
-                    .fetch_add(thread_metrics.collect_us.load(Relaxed), Relaxed);
-            }
+                .reduce(
+                    CollectRentInPartitionInfo::default,
+                    CollectRentInPartitionInfo::reduce,
+                );
 
+            // bprumo TODO: is it safe to move this "release-the-hold" up here? No reason to hold the
+            // range while we write the results/metrics, right?
             self.rc
                 .accounts
                 .hold_range_in_memory(&subrange_full, false, thread_pool);
```

Comment:
I moved these lines up earlier. It looks like we can release the hold ASAP, but want to make sure. Is this safe?